schema/snapcraft.yaml

```diff
@@ -123,6 +123,11 @@ properties:
             uniqueItems: true
             items:
               type: string
+          aliases:
+            type: array
+            items:
+              type: string
+              pattern: "^[a-zA-Z0-9][-_.a-zA-Z0-9]*$"
```

Comment:
Alright, Gustavo got back to me about why these are different (wanting to support lower- and upper- case, numbers, underscores, periods, etc. in aliases) but he did say that they shouldn't end in a hyphen, underscore, or period.